packages/menu/src/MenuItem.ts

```diff
@@ -276,6 +324,108 @@ export class MenuItem extends LikeAnchor(Focusable) {
         }
     }
 
+    public closeOverlay?: (leave?: boolean) => Promise<void>;
+
+    protected handleSubmenuClick(): void {
+        this.openOverlay({ immediate: true });
+    }
+
+    protected handlePointerenter(): void {
+        if (this.leaveTimeout) {
+            clearTimeout(this.leaveTimeout);
+            delete this.leaveTimeout;
+        } else {
+            this.openOverlay();
+        }
+    }
+
+    protected leaveTimeout?: ReturnType<typeof setTimeout>;
+
+    protected handlePointerleave(): void {
+        if (this.hasSubmenu && this.open) {
+            this.active = true;
+            this.leaveTimeout = setTimeout(() => {
+                delete this.leaveTimeout;
+                if (this.closeOverlay) this.closeOverlay(true);
+            }, POINTERLEAVE_TIMEOUT);
+        }
+    }
+
+    protected handleSubmenuChange = (): void => {
+        this.click();
```

Comment:
If this is equivalent to `this.openOverlay({ immediate: true });`, it might be clearer to call that directly instead of simulating a click. Or is a click event & bubbling necessary to be listened to higher up?